### PR TITLE
githooks for spacing issues

### LINKED ISSUE #4582: Coding standard
Align coding standards across PRs.
Create a linter that automatically formats everyone's code to the same code style. For example, Warp uses this for TypeScript: https://github.com/NethermindEth/warp/blob/develop/.eslintrc.json. 
They also have what’s a called a ‘pre-commit hook’ which means every time we ‘git commit’ it automatically formats their code to some agreed code formatting.

### PR DESCRIPTION
**Pull request by smartprogrammer93**
Resolves #4582

With the suggested approach above, users need to : 
1. install dotnet format
```
dotnet tool install -g dotnet-format
``` 
2. assign the hooksPath like so
```
git config --local core.hooksPath .githooks/
```

## Changes:
-Adding Hook to call dotnet format

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ ] Yes
- [x] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [x] No

### CODE REVIEW

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+dotnet format whitespace src/Nethermind/ --folder
```
Comment (asdacap) [inline .githooks/pre-commit:3]: Have you check if this work with windows?

Comment (smartprogrammer93) [reply .githooks/pre-commit:3]: Yes i actually use windows machine. you just need to install dotnet format like in the description of the PR

**Review by asdacap — APPROVED** [review]
Checked semi-manulally with git diff -w. LGTM.

**Comment by DemuirGos** [comment]
Checked with this [script](https://gist.github.com/DemuirGos/66dc441859d1c408e40b8caff3519f3e). LGTM

**Comment by dceleda** [comment]
Ok, let's merge it next week to master just after the release branch out.

**Pull request merged**